docs/sources/flow/reference/config-blocks/declare.md

```diff
@@ -0,0 +1,82 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/config-blocks/declare/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/config-blocks/declare/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/config-blocks/declare/
+- /docs/grafana-cloud/send-data/agent/flow/reference/config-blocks/declare/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/config-blocks/declare/
+description: Learn about the declare configuration block
+menuTitle: declare
+title: declare block
+---
+
+# declare block
+
+`declare` is an optional configuration block used to define a new [custom component][].
+`declare` blocks must be given a label that determines the name of the custom component.
+
+[custom component]: {{< relref "../../concepts/custom_components.md" >}}
+
+## Example
+
+```river
+declare "COMPONENT_NAME" {
+    COMPONENT_DEFINITION
+}
+```
+
+## Arguments
+
+The `declare` block has no predefined schema for its arguments; the body of the
+`declare` block is used as the component definition. The body may contain:
+
+* [argument][] blocks
+* [export][] blocks
+* [declare][] blocks
+* [import][] blocks
+* Component definitions (either built-in or custom components)
+
+The `declare` block may not contain any configuration blocks that are not
+listed above.
+
+[argument]: {{< relref "./argument.md" >}}
+[export]: {{< relref "./export.md" >}}
+[declare]: {{< relref "./declare.md" >}}
+[import]: {{< relref "../../concepts/modules.md#importing-modules" >}}
```

Comment:
Another one... we should be defining these as links not relrefs (throughout all new content, not just this spot).  See the warning here: https://grafana.com/docs/writers-toolkit/write/shortcodes/#relref

Since this is new content... https://grafana.com/docs/writers-toolkit/write/links/

I'll refactor the existing docs in an upcoming release to remove relrefs. If we keep these relrefs in this PR it's fine. I can catch it in the general cleanup planned for after v0.40.

Reply:
thanks for the warning, I will keep them for now because we want to merge it before the cut